### PR TITLE
Update graphql_client.ex

### PR DESCRIPTION
**Pull request by certainty**
Just a little restructuring. I haven't tried to compile it so it might contain errors

### CODE REVIEW

File: lib/graphql_client.ex [hunk 1]
```diff
@@ -3,33 +3,11 @@ defmodule GraphqlClient do
...
-       {:ok, %HTTPoison.Response{status_code: 403, body: body, headers: headers}} ->
-         IO.puts "forbidden"
-         IO.puts body
-       {:ok, %HTTPoison.Response{status_code: 200, body: body, headers: headers}} ->
-         IO.puts "success"
-         IO.inspect Poison.decode!(body)
-       {:error, %HTTPoison.Error{reason: reason}} ->
-         IO.inspect reason
-     end
+    with options_               <- Keyword.merge(default_options, options),
```
Comment (scepticulous) [inline lib/graphql_client.ex]: I like that :+1:

File: lib/graphql_client.ex [hunk 2]
```diff
@@ -38,14 +16,22 @@ defmodule GraphqlClient do
     )
     options
   end
-
-  defp default_env do
-    "dev"
+  
+  defp process_response({:ok, %HTTPoison.Response{status_code: 403, body: body, headers: _headers}}) do
```
Comment (scepticulous) [inline lib/graphql_client.ex]: Great Idea to move the pattern matching into functions. way better than using a case statement.
:+1:

File: lib/graphql_client.ex [hunk 3]
```diff
@@ -3,33 +3,11 @@ defmodule GraphqlClient do
...
-       {:ok, %HTTPoison.Response{status_code: 200, body: body, headers: headers}} ->
-         IO.puts "success"
-         IO.inspect Poison.decode!(body)
-       {:error, %HTTPoison.Error{reason: reason}} ->
-         IO.inspect reason
-     end
+    with options_               <- Keyword.merge(default_options, options),
+         [ headers | [ data ] ] <- parse_config(options_[:env]),
+         full_url               <- to_string(data[:host]) <>  options_[:url],
+    do: process_response(HTTPoison.get(full_url, headers)
```
Comment (scepticulous) [inline lib/graphql_client.ex]: there is a ) missing at the end of the line.

**Comment by scepticulous** [comment]
After adding the missing )
compilation fails with:

> lib/graphql_client.ex:7: warning: this clause cannot match because a previous clause at line 7 always matches
lib/graphql_client.ex:9: warning: this clause cannot match because a previous clause at line 9 always matches
Compiled lib/graphql_client.ex
== Compilation error on file lib/schema.ex ==
** (Protocol.UndefinedError) protocol String.Chars not implemented for {{:., [line: 3], [{:__aliases__, [line: 3], [:GraphqlClient, :Schema]}, :types]}, [line: 3], []}
    (elixir) lib/string/chars.ex:3: String.Chars.impl_for!/1
    (elixir) lib/string/chars.ex:17: String.Chars.to_string/1
    lib/absinthe/schema/notation.ex:745: Absinthe.Schema.Notation.do_import_types/2
    expanding macro: Absinthe.Schema.Notation.import_types/1
    lib/schema.ex:3: GraphqlClient.Schema (module)

**Comment by scepticulous** [comment]
Just want to thank you for the work you put into this.

**Comment by certainty** [comment]
Does it compile now?

**Comment by certainty** [comment]
The 'with' seems to be problematic. I should probably try to compile it and see for myself 😊

**Comment by scepticulous** [comment]
I implemented most of the pull request myself, so that I dont have to fix the compiler errors and still get all the improvements I fully understand.
See: https://github.com/scepticulous/elixir-graphql-client/commit/dfb051f5cad1a24d76cbaa9111a235e5d6c54bef

**Pull request closed without merging**